### DISCUSSION TITLE
How to capture data from a response and use it in a request?

### DISCUSSION THREAD

**Discussion by wiliamvj** [thunderclient/thunder-client-support#1464] [Q&A] [answered]
I have a login route that returns a token, I would like to capture the token and use it in routes that are authenticated

Something similar to this:
![image](https://github.com/rangav/thunder-client-support/assets/30341785/a0177ae4-d088-466c-a7c0-5f285964001e)

insomnia does this
![image](https://github.com/rangav/thunder-client-support/assets/30341785/0d43af16-b0da-4cc5-9ea7-ba8b55c6957a)

**Comment by rangav** [comment, marked as answer]
Hi @wiliamvj 

- Use the Tests tab and select Set Env Variable test to save response data to a variable
- Then use that variable in next requests using {{varName}}
